packages/curve-ui-kit/src/shared/ui/LargeTokenInput.tsx

```diff
@@ -281,23 +290,58 @@ export const LargeTokenInput = <T extends Amount>({
     maxBalance?.onMax?.call(null)
   }, [handlePercentageChange, maxBalance?.onMax])
 
+  const componentId = useId()
+
   return (
     <Stack
+      id={componentId}
       data-testid={testId}
-      gap={Spacing.xs}
+      gap={Spacing.sm}
       sx={{
         backgroundColor: (t) => t.design.Inputs.Large.Default.Fill,
-        padding: Spacing.md,
         outline: (t) =>
           `1px solid ${isError ? t.design.Layer.Feedback.Error : t.design.Inputs.Base.Default.Border.Default}`,
       }}
     >
-      <Stack gap={Spacing.xs}>
-        {/** First row is an optional label describing the input */}
-        {label && (
-          <Typography variant="bodyXsRegular" color="textSecondary">
-            {label}
-          </Typography>
+      <Stack gap={Spacing.xs} sx={{ padding: Spacing.sm }}>
+        {/** First row is an optional label describing the input and/or chips */}
+        {(label || maxBalance?.showChips) && (
+          <Stack direction="row" alignItems="center">
+            {label && (
+              <Typography variant="bodyXsRegular" color="textSecondary">
+                {label}
+              </Typography>
+            )}
+
+            {maxBalance?.showChips && (
+              <Stack
+                direction="row"
+                gap={Spacing.xxs}
+                sx={{
+                  flexGrow: 1,
+                  justifyContent: 'end', // Hide by default, show on parent hover
```

Comment:
is this comment in the right line? :thinking:

Reply:
woops!